Add spec and proofs for load64 and store64

manastasova opened the pull request:
This PR adds the spec and proofs for load64 and store64.

Notes::
	- Change the signature of function `store64` from `static void store64(uint8_t x[8], uint64_t u)` to `static void store64(uint8_t x[8], const uint64_t u)`, marking the input as a `const`.
	-  Add explicit `& 0xFF` before casting `uint64_t u` to `uint8_t x[i]` to prevent overlflows

hanno-becker commented on mldsa/fips202/fips202.c:
```diff
@@ -46,13 +52,21 @@ static uint64_t load64(const uint8_t x[8])
+  )
   {
-    x[i] = u >> 8 * i;
+    /* Make sure that u >> (8 * i) will not overflow uint8_t x[i] */
```
Suggestion: 
```suggestion
    /* Explicitly truncate to uint8_t */
```

hanno-becker commented on mldsa/fips202/fips202.c:
```diff
@@ -46,13 +52,21 @@ static uint64_t load64(const uint8_t x[8])
  *              - uint64_t u: input 64-bit unsigned integer
  **************************************************/
-static void store64(uint8_t x[8], uint64_t u)
+static void store64(uint8_t x[8], const uint64_t u)
```
`const` is not too common for arguments passed by value, not by reference. The qualification only restricts what the implementation can do, but it has no impact on the caller.

hanno-becker commented on proofs/cbmc/store64/store64_harness.c:
```diff
@@ -0,0 +1,13 @@
+
+#include "fips202/fips202.h"
+
+extern void store64(uint8_t x[8], const uint64_t u);
```
As mentioned above, the `const` has no meaning for external signature because `u` is passed by value.

hanno-becker reviewed (changes requested):
Thanks @manastasova! Can you remove the `const` annotation and slightly clarify one of the comments, as indicated? Otherwise LGTM

manastasova commented:
> Thanks @manastasova! Can you remove the `const` annotation and slightly clarify one of the comments, as indicated?
> 
> Also, can you squash the commits, and make sure that lines in the body of the commit message are no longer than ~75 characters per line?
> 
> Otherwise LGTM

Thanks a lot @hanno-becker! Just addressed your comments.

hanno-becker reviewed (changes requested):
Thank you @manastasova, the code looks good now! Can you please fix the commit message (no lines > 75 characters, typos)? Then we're good to go.
